File: Android/iOrder/app/src/main/java/hr/fer/oobl/iorder/iorder/injection/activity/ActivityComponent.java
package hr.fer.oobl.iorder.iorder.injection.activity;

import dagger.Component;
import hr.fer.oobl.iorder.iorder.injection.activity.modules.ActivityModule;
import hr.fer.oobl.iorder.iorder.injection.activity.modules.ActivityPresenterModule;
import hr.fer.oobl.iorder.iorder.injection.application.ApplicationComponent;

@ActivityScope
@Component(
        dependencies = {
                ApplicationComponent.class
        },
        modules = {
                ActivityModule.class,
                ActivityPresenterModule.class
        }
)
public interface ActivityComponent extends ActivityComponentInjects, ActivityComponentExposes {

    final class Initializer {

        static public ActivityComponent init(final DaggerActivity daggerActivity, final ApplicationComponent applicationComponent) {
            return DaggerActivityComponent.builder()
                                          .applicationComponent(applicationComponent)
                                          .activityModule(new ActivityModule(daggerActivity))
                                          .activityPresenterModule(new ActivityPresenterModule(daggerActivity))
                                          .build();
        }

        private Initializer() {

        }
    }
}

File: Android/iOrder/app/src/main/java/hr/fer/oobl/iorder/iorder/injection/activity/ActivityComponent.kt
package hr.fer.oobl.iorder.iorder.injection.activity

import dagger.Component
import hr.fer.oobl.iorder.iorder.injection.activity.modules.ActivityModule
import hr.fer.oobl.iorder.iorder.injection.activity.modules.ActivityPresenterModule
import hr.fer.oobl.iorder.iorder.injection.application.ApplicationComponent

@ActivityScope
@Component(dependencies = arrayOf(ApplicationComponent::class), modules = arrayOf(ActivityModule::class, ActivityPresenterModule::class))
interface ActivityComponent : ActivityComponentInjects, ActivityComponentExposes {

    object Initializer {

        fun init(daggerActivity: DaggerActivity, applicationComponent: ApplicationComponent): ActivityComponent {
            return DaggerActivityComponent.builder()
                    .applicationComponent(applicationComponent)
                    .activityModule(ActivityModule(daggerActivity))
                    .activityPresenterModule(ActivityPresenterModule(daggerActivity))
                    .build()
        }
    }
}